sparselsh/lsh.py

```diff
@@ -326,12 +337,15 @@ def query(self, query_points, num_results=None, distance_func=None,
             raise ValueError(
                 "The distance function %s is invalid." % distance_func
             )
-        if dist_threshold is not None and \
-            (dist_threshold <= 0 or (distance_func == "cosine" and dist_threshold > 1.0)):
+        if dist_threshold and (not isinstance(dist_threshold, int) and \
+            not isinstance(dist_threshold, float)) and \
+            (dist_threshold <= 0 or \
+                (distance_func == "cosine" and dist_threshold > 1.0)):
             raise ValueError(
                 "The distance threshold %s is invalid." % dist_threshold
             )
-        if num_results is not None and num_results < 1:
+        if num_results and not isinstance(num_results, int) and \
+            num_results < 1:
```

Comment:
Revert. We don't need to check for int.

Reply:
Fixed by converting this to an assertion check at the beginning of the function, which is much simpler and easier to read.